test/integration/ads_integration_test.cc

```diff
@@ -1749,149 +1749,172 @@ TEST_P(AdsIntegrationTest, ContextParameterUpdate) {
-      Config::TypeUrl::get().Listener, {},
-      {
-          buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
-                        "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
-                        route_name_0),
-          // Ignore resource in impostor namespace.
-          buildListener("xdstp://test/envoy.config.listener.v3.Listener/impostor/"
-                        "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
-                        route_name_0),
-          // Ignore non-matching context params.
-          buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
-                        "baz?xds.node.cluster=cluster_name&xds.node.id=other_name",
-                        route_name_0),
-      },
-      {}, "1");
-
-  EXPECT_TRUE(
-      compareDiscoveryRequest(Config::TypeUrl::get().ClusterLoadAssignment, "1", {}, {}, {}));
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "", {},
-                                      {route_name_0}, {}));
+  const auto listeners = {
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
+                    "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
+                    route_name_0),
+      // Ignore resource in impostor namespace.
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/impostor/"
+                    "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
+                    route_name_0),
+      // Ignore non-matching context params.
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
+                    "baz?xds.node.cluster=cluster_name&xds.node.id=other_name",
+                    route_name_0),
+  };
+  sendDiscoveryResponse<envoy::config::listener::v3::Listener>(Config::TypeUrl::get().Listener,
+                                                               listeners, listeners, {}, "1");
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().ClusterLoadAssignment, "1", {}, {}, {},
+                                      /*expect_node=*/isSotw()));
+
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "", {route_name_0},
+                                      {route_name_0}, {}, /*expect_node=*/isSotw()));
+  const auto route_config = buildRouteConfig(route_name_0, cluster_name);
   sendDiscoveryResponse<envoy::config::route::v3::RouteConfiguration>(
-      Config::TypeUrl::get().RouteConfiguration, {}, {buildRouteConfig(route_name_0, cluster_name)},
-      {}, "1");
+      Config::TypeUrl::get().RouteConfiguration, {route_config}, {route_config}, {}, "1");
 
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "1", {}, {}, {}));
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "1", {}, {}, {}));
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "1", {}, {}, {},
```

Comment:
I'm not sure why node is being sent. I think you can set `ads_config->set_set_node_on_first_message_only(true);` in the initialization phase.